### DISCUSSION TITLE
Is it OK for fregrid_parallel to select different but equivalent parent tiles under certain circumstances.

### DISCUSSION THREAD

**Discussion by ngs333** [NOAA-GFDL/FRE-NCtools#250] [General]
In producing weights file, fregrid_parallel may not be recording the same tile number (field tile number in mosaic) 
as the serial fregrid.  This can be seen when mapping cubic sphere to lat-lon grid, but (obviously) not when
mapping from lat-lon to lat-lon.  Additionally, applying the differing weight files to remapping does not seem to make
a difference in the actual values of the fields. E.g. NCTools Test15 produces the same final output, but upon inspection it can be seen that the produced weights files are not identical. Additionally, one can observe that ncells (or index length or dimention) is the same in both serial and parallel cases. This result can be seen in the baseline and older code (to the extent tested) and does not depend on installation or compiler, etc.

It may be that what is happening is that in creating the x-grid, the serial fregrid and a run of the parallel fregrid
 sometimes choose different tiles (of the cubic sphere) for equivalent cells. Of course, this could only occur at tile boundaries : where two source cubic sphere tiles meet and in such a way where the same overlap to a target call exist in each. If such is the case, it is it desirable to force an ordering of the tiles used in the xgrid creation for the sake of "perfect" agreement between the serial and parallel versions?

**Comment by ngs333** [comment]
@mlee03 @ceblanton @bensonr @rem1776 @nikizadehgfdl 
Just adding these folks to the conversation as a discussion already started.

**Comment by jwdGFDL** [comment]
This situation has existed literally for decades. It is something that we should document and probably spend some face time with key individuals (Sergey, maybe Alistair, etc) to determine whether there's further action to be taken. However, it seems to be a bit off track since I thought the goal of the work was to evaluate xESMF as a potential replacement.

**Reply by ngs333** [reply]
@jwdGFDL Thanks for the info. This arose from GPU vs baseline comparison Mikyung is doing.

**Reply by bensonr** [reply]
I see this as a non-issue and not worth mentioning beyond some documentation in a readme or release note.  It's a situation that has existed for years with no one noting it previously.
